source/precharge/main.c

```diff
@@ -227,21 +240,44 @@ void heartbeatTask()
     PHAL_toggleGPIO(HEARTBEAT_LED_GPIO_Port, HEARTBEAT_LED_Pin);
 }
 
+//Original monitorStatus() function
+// void monitorStatus()
+// {
+//     uint16_t err = 0;
+
+//     PHAL_writeGPIO(ERROR_LED_GPIO_Port, ERROR_LED_Pin, !PHAL_readGPIO(IMD_STATUS_GPIO_Port, IMD_STATUS_Pin) | err);
+//     PHAL_writeGPIO(BMS_STATUS_GPIO_Port, BMS_STATUS_Pin, err ? 0 : 1);
+
+// }
 void monitorStatus()
 {
     uint16_t err = 0;
 
     PHAL_writeGPIO(ERROR_LED_GPIO_Port, ERROR_LED_Pin, !PHAL_readGPIO(IMD_STATUS_GPIO_Port, IMD_STATUS_Pin) | err);
-    PHAL_writeGPIO(BMS_STATUS_GPIO_Port, BMS_STATUS_Pin, err ? 0 : 1);
+    switch(err) {
```

Comment:
I don't believe a switch case is necessary in this context. Ensure the error takes into account Temperature, Orion status, and Orion timeout.